clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1521,6 +1522,25 @@ void CodeGenFunction::GenerateCode(GlobalDecl GD, llvm::Function *Fn,
   // Emit the standard function prologue.
   StartFunction(GD, ResTy, Fn, FnInfo, Args, Loc, BodyRange.getBegin());
 
+  SyclOptReportHandler &OptReportHandler = CGM.getDiags().OptReportHandler;
+  if (OptReportHandler.HasOptReportInfo(FD)) {
+    llvm::OptimizationRemarkEmitter ORE(Fn);
+    int Count = 0;
+    for (auto &ORI : OptReportHandler.GetInfo(FD)) {
```

Comment:
Rather than keep a separate count, I believe you can use `llvm::enumerate(Container)` to handle this automagically for you. e.g.,
```
for (auto X : llvm::enumerate(*ORI)) {
  X.value(); // Gives you the value of the iteration.
  X.index(); // Gives you the index of the iteration.
}
```